src/validation.cpp

```diff
@@ -1876,25 +1871,19 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
         // * witness (when witness enabled in flags and excludes coinbase)
         nSigOpsCost += GetTransactionSigOpCost(tx, view, flags);
         if (nSigOpsCost > MAX_BLOCK_SIGOPS_COST)
-            return state.Invalid(ValidationInvalidReason::CONSENSUS, error("ConnectBlock(): too many sigops"),
+            return state.Invalid(BlockValidationResult::CONSENSUS, error("ConnectBlock(): too many sigops"),
                              REJECT_INVALID, "bad-blk-sigops");
 
         txdata.emplace_back(tx);
         if (!tx.IsCoinBase())
         {
             std::vector<CScriptCheck> vChecks;
             bool fCacheResults = fJustCheck; /* Don't cache results if we're actually connecting blocks (still consult the cache, though) */
-            if (!CheckInputs(tx, state, view, fScriptChecks, flags, fCacheResults, fCacheResults, txdata[i], nScriptCheckThreads ? &vChecks : nullptr)) {
-                if (state.GetReason() == ValidationInvalidReason::TX_NOT_STANDARD) {
```

Comment:
In commit "[validation] Add CValidationState subclasses" (b31987c45045cc26ca7867d9d8aa645f5d36b5e6)

Similar questions here as for the CheckTxInputs check above. What's the explanation for why there's no change in behavior here? Would CheckInputs never set reasons other than CONSENSUS and TX_NOT_STANDARD?

And is it a problem that the recent_consensus code will be translated to just consensus?

Reply:
> Would CheckInputs never set reasons other than CONSENSUS and TX_NOT_STANDARD?

Exactly correct.

> And is it a problem that the recent_consensus code will be translated to just consensus?

As above. Any PR that introduces recent_consensus should include tests to make sure it does what it's supposed to.